tests/rag/test_vector_store_pgvector.py

```
@@ -0,0 +1,124 @@
+
+def test_pgvector_store_validates_constructor_args(fake_connect: list[_FakeConnection]) -> None:
+    with pytest.raises(ValueError):
+        PgVectorStore(dsn="", index_name="documents")
+    with pytest.raises(ValueError):
+        PgVectorStore(dsn="postgresql://u:p@localhost:5432/db", index_name="bad-name")
+    with pytest.raises(ValueError):
+        PgVectorStore(dsn="postgresql://u:p@localhost:5432/db", embedding_dim=0)
+
+
+def test_pgvector_search_returns_normalized_docs(
+    monkeypatch: pytest.MonkeyPatch, fake_connect: list[_FakeConnection]
```

Ruff (0.15.0)

[warning] Unused function argument: `fake_connect`

(ARG001)